### PR TITLE
Support switch expression in CompleteStatementHandler

### PR DESCRIPTION
**Pull request by Cosifne**
Fix https://github.com/dotnet/roslyn/issues/57213

After this change, if you enter `;` inside the `{   }` of switch expression, the caret would be moved to the end of the statement.

### CODE REVIEW

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs
Comment (Youssef1313) [inline src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs:487]: object/collection initializers?

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs
Comment (Youssef1313) [inline src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs:162]: Why this is different than above code that calculates brackets and parentheses? The above just calls `startingNode.GetX`

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs
Comment (Youssef1313) [inline src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs:204]: object/collection/with initializers?

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs
Comment (Cosifne) [reply src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs]: I do test initializers (if you see the comment history)
However, there are many tests failing, like https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs,1704
So I decided to start with switch expression only

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs
Comment (Cosifne) [reply src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs]: Changed : )

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs
Comment (Cosifne) [reply src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs]: The original reason is that I feel it might make sense to support
```
var x = switch bar
{
   "Case1" $$=> 1,
    _ => 3,
}
```
And move the caret to the end. 
But yeah this is too board, I have deleted this part.

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs
Comment (Cosifne) [reply src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs:204]: Same as the comment aboved.

**Comment by Cosifne** [comment]
Tag @CyrusNajmabadi  and @Youssef1313  to have a second review : )

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by Youssef1313 — APPROVED** [review]
LGTM

**Pull request merged**
